org.osgi.test.junit5.cm/src/test/java/org/osgi/test/junit5/cm/ConfigAnnotationNested.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.cm;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.Dictionary;
+import java.util.Map;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.osgi.service.cm.Configuration;
+import org.osgi.test.common.annotation.config.InjectConfiguration;
+import org.osgi.test.common.annotation.config.WithConfiguration;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+@ExtendWith(ConfigurationExtension.class)
+@WithConfiguration(pid = "class")
+@TestMethodOrder(OrderAnnotation.class)
+@Nested
+public class ConfigAnnotationNested {
+
+	@Test
+	@WithConfiguration(pid = "class.method1")
+	public void test_class_method1(@InjectConfiguration("class") Configuration c_class,
+		@InjectConfiguration("class.method1") Configuration c_class_Method1,
+		@InjectConfiguration("class.inner1") Configuration c_class_Inner1,
+		@InjectConfiguration("class.inner1.method1") Configuration c_class_Inner1_Method1,
+		@InjectConfiguration("class.inner1.method2") Configuration c_class_Inner1_Method2,
+		@InjectConfiguration("class.inner2") Configuration c_class_Inner2,
+		@InjectConfiguration("class.inner2.method1") Configuration c_class_Inner2_Method1) throws Exception {
+		assertNotNull(c_class);
```

Comment:
All the tests in this project would be much better if we used assertj assertions to test.

Reply:
And for tests like this one that have multiple assertions, they'd be even better if we used soft assertions. That one one failure doesn't stop the rest of the assertions from running & giving feedback to the developer.

Reply:
done